auth/auth.go

```diff
@@ -0,0 +1,157 @@
+package auth
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/influxdata/influxdb/influxql"
+	"github.com/pkg/errors"
+)
+
+// Interface for authenticating and retrieving users.
+type Interface interface {
```

Comment:
Not crazy about using `Interface` as the interface name. Reads okay in the code but feels weird. Also, if the `auth` package ever needed a second interface for something, it would end up with a regular name, which would be inconsistent.